_posts/2022-10-20-datafusion-13.0.0.md

```diff
@@ -0,0 +1,226 @@
+
+
+
+# DataFusion in Action
+
+While DataFusion really shines as an embeddable query engine, if you want to try it out and get a feel for its power, you can use the basic[`datafusion-cli`](https://docs.rs/datafusion-cli/13.0.0/datafusion_cli/) tool to get a sense for what is possible to add in your application
+
+(TODO example here of using datafusion-cli to query from local parquet files on disk)
+
+TODO: also mention you can use the same thing to query data from S3
+
+
+
+# Summary
+
+We have increased the frequency of DataFusion releases to monthly instead of quarterly. This
+makes it easier for the increasing number of projects that now depend on DataFusion.
+
+We have also completed the "graduation" of [Ballista to its own top-level arrow-ballista repository](https://docs.google.com/document/d/1jNRbadyStSrV5kifwn0khufAwq6OnzGczG4z8oTQJP4/edit?usp=sharing)
+which decouples the two projects and allows each project to move even faster.
+
+Along with numerous other bug fixes and smaller improvements, here are some of the major advances:
+
+## Advanced SQL
+- Support for GROUPING SETS/CUBE/ROLLUP (#2716)
+- Custom window frame logic (support `ROWS`, `RANGE`, `PRECEDING` and `FOLLOWING` for window functions) (#3570)
+- Add support for correlated subqueries (#2885)
+- `ROLLUP` and `CUBE` grouping set expressions (#2446)
+- #2405 SUM DISTINCT aggregate support Sum distinct support
+- Support for `IN` and `NOT IN` Subqueries by rewriting them to `SEMI` / `ANTI` (#2421)
+- Support for non equality predicates in  `ON` clause of  `LEFT`, `RIGHT, `and `FULL` joins (#2591)
+- Exact `MEDIAN` (#3009)
+
+# More DDL Support
+ - `CREATE VIEW` (#2279)
+ - `DESCRIBE <table>` (#2642)
+ - Custom / Dynamic table provider factories (#3311)
+ - `SHOW CREATE TABLE` for support for views (#2830)
+
+# Faster Execution
+ - Optimizations of TopK (queries with a `LIMIT` or `OFFSET` clause):  (#3527), (#2521)
+ - Reduce `left`/`right`/`full` joins to `inner` join (#2750)
+ - Convert  cross joins to inner joins when possible (#3482)
+ - Sort preserving `SortMergeJoin` (#2699)
+ - Improvements in group by and sort performance (#2375)
+
+# Optimizer Enhancements
+- Casting / coercion now happens logical planning (#3185) (#3396) (#3636)
+- More sophisticated expression analysis and simplification
+
+# Parquet
+ - Parquet reader can now read directly from remote parquet files on object storage (#2489) (#2677) (#3051)
+ - Experimental support for “predicate pushdown” that implements late materialization after filtering during the scan (we plan another blog post on this soon).
+ - Support reading directly from S3 via `datafusion-cli ` (#3631)
+
+# DataType Support
+- Support for `IN` LIST,  support, casting, etc for `Decimal`
+- Timestamps: `date_bin` built-in function (#3034), timestamp plus minus interval (#3110), TIME literal values (#3010)
+- Binary operations (AND, XOR, etc):  (#3037) (#1619) (#3420) (#3430)
+
```

Comment:
- `IS TRUE/FALSE` and `IS [NOT] UNKNOWN` (#3235), (#3246)

Reply:
- Support TimestampTz ( apache/arrow-datafusion#3660 )

Reply:
In d02bb30657 👍